session-replay/Sources/DatadogSessionReplay/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeRecorders/UINavigationBarRecorder.swift

```diff
@@ -0,0 +1,73 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import UIKit
+
+internal struct UINavigationBarRecorder: NodeRecorder {
+    func semantics(of view: UIView, with attributes: ViewAttributes, in context: ViewTreeSnapshotBuilder.Context) -> NodeSemantics? {
+        guard let navigationBar = view as? UINavigationBar else {
+            return nil
+        }
+
+        let builder = UINavigationBarWireframesBuilder(
+            wireframeRect: inferOccupiedFrame(of: navigationBar, in: context),
+            wireframeID: context.ids.nodeID(for: navigationBar),
+            attributes: attributes,
+            color: inferColor(of: navigationBar)
+        )
+
+        return SpecificElement(wireframesBuilder: builder, recordSubtree: true)
+    }
+
+    private func inferOccupiedFrame(of navigationBar: UINavigationBar, in context: ViewTreeSnapshotBuilder.Context) -> CGRect {
+        // TODO: RUMM-2791 Enhance appearance of `UITabBar` and `UINavigationBar` in SR
+        var occupiedFrame = navigationBar.frame
+        for subview in navigationBar.subviews {
+            let subviewFrame = subview.convert(subview.bounds, to: context.coordinateSpace)
+            occupiedFrame = occupiedFrame.union(subviewFrame)
+        }
+        return occupiedFrame
+    }
```

Comment:
We need to infer the frame, because the actual `frame` of `UINavigationBar` doesn't include extended edges. These are denoted by its subviews:

<img width="839" alt="Screenshot 2022-12-07 at 19 53 24" src="https://user-images.githubusercontent.com/2358722/206270572-23492c16-5327-4b33-8d0c-0ca842924100.png">